sycl/plugins/esimd_emulator/CMakeLists.txt

```diff
@@ -45,42 +61,23 @@ if (DEFINED CM_LOCAL_SOURCE_DIR)
       BUILD_BYPRODUCTS  ${LIBCM} ${LIBIGFXCMRT_EMU}
     )
   endif()
-else ()
-  if (DEFINED CM_PACKAGE_URL)
-    # Downloading pre-built CM Package
-    file (MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_install)
-    ExternalProject_Add(cm-emu
-      URL               ${CM_PACKAGE_URL}
-      CONFIGURE_COMMAND ""
-      BUILD_COMMAND     ""
-      UPDATE_COMMAND    ""
-      INSTALL_DIR       ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_install
-      INSTALL_COMMAND   ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_BINARY_DIR}/cm-emu-prefix/src/cm-emu/ <INSTALL_DIR>
-      BUILD_BYPRODUCTS  ${LIBCM} ${LIBIGFXCMRT_EMU}
-    )
-  else()
-    # Build from CM source tree fetched from github
-    if (MSVC)
-      message(FATAL_ERROR "Online-building of CM_EMU library is not supported under Windows environment")
-    else()
-      # Arguments for online patching to suppress log message from CM_EMU
-      # Replacing CM_EMU's log print-out macro controlled by 'GFX_EMU_WITH_FLAGS_'
-      # with blank space from $CM_EMU_SRC/common/emu_log.h
-      set (replacing_pattern s/{\ ?GFX_EMU_WITH_FLAGS_.*//g)
-      ExternalProject_Add(cm-emu
-        GIT_REPOSITORY    https://github.com/intel/cm-cpu-emulation.git
-        GIT_TAG           f9d167edbcb995e713d5a8
-        BINARY_DIR        ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_build
-        INSTALL_DIR       ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_install
-        PATCH_COMMAND     perl -pi.back -e ${replacing_pattern} ${CMAKE_CURRENT_BINARY_DIR}/cm-emu-prefix/src/cm-emu/common/emu_log.h
-        CMAKE_ARGS        -DLIBVA_INSTALL_PATH=/usr
-                          -D__SYCL_EXPLICIT_SIMD_PLUGIN__=true
-                          -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
-        BUILD_BYPRODUCTS  ${LIBCM} ${LIBIGFXCMRT_EMU}
-      )
-    endif()
+else()
+  set(ACTUAL_CM_EMU_PREBUILT_PACKAGE ${DEFAULT_CM_EMU_PREBUILT_PACKAGE})
+  if (DEFINED USE_CM_EMU_PREBUILT_PACKAGE)
+    set(ACTUAL_CM_EMU_PREBUILT_PACKAGE ${USE_CM_EMU_PREBUILT_PACKAGE})
   endif()
-endif ()
+  file (MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/cm-emu_install)
+  ExternalProject_Add(cm-emu
```

Comment:
`ExternalProject_Add` seems not the right tool for downloading pre-built packages, AFAIU, it is for adding external CMakeProjects and build them from source. Please check

Reply:
`FetchContent_*` functions can be used for downloading archives. However, it seems that using the functions requires extra steps for CM_EMU as the downloaded library package is used for both toolchain build and kernel compilation. I would like to keep this part as it is with TODO comments due to urgency of this change set. CMake for LevelZero PI also uses `ExternalProject_Add`  for LevelZero module and already has TODO comment for `FetchContent` - https://github.com/intel/llvm/blob/863383bac8eeab168f39f4da8e94df860458cbf0/sycl/plugins/level_zero/CMakeLists.txt#L30.